portfolio/src/main/webapp/script.js

```diff
@@ -50,26 +83,48 @@ async function toggleCommentHistorySection() {
  * Fetches and adds a history of comments, and the theoretical maximum and
  * default number of comments, the total number of pages, and the current
  * page ID, to the page.
- * Backup and restore comment history as necessary according to the disabling
- * and enabling of the comment history section.
+ * Backup and restore comment submission/history sections as necessary
+ * according to the disabling and enabling of the comment history section,
+ * and whether the user is logged in.
  * Show a warning if the user input value of the number of comments to display
  * or page ID is invalid, or if the latest user input may be a XSS attack.
  */
 async function addComments(pageId) {
+  const commentSubmissionSection = document
+      .getElementById('comment-submission-section');
   const commentHistorySection = document
       .getElementById('comment-history-section');
-  // If the comment history section was disabled just now, backup the current
-  // content and remove that content from the page.
-  if (!enableCommentHistorySection) {
+  await getAuthentication();
+
+  // If the comment history section was disabled just now or if the user is
+  // not logged in, backup the current comment history content and remove
+  // that content from the page. Additionally, if the user is not logged in,
+  // backup and remove the comment submission form.
+  if (!enableCommentHistorySection || !isUserLoggedIn) {
```

Comment:
Can you rewrite this if else logic? You're checking the same thing e.g. isUserLoggedIn multiple times. Perhaps think about separating out the logic into 2 helper functions: one for comment history section and one for comment submission section?

Reply:
Got it, thank you!

Reply:
I separated out the logic for the comment history section and the comment submission section. I also separated out the logic for fetching comment history.
Also, unlike the comment history section, the comment submission form always looks the same. So, I removed the extra logic for backing up & restoring the submission form, and changed the code to directly hide & display the submission form.